### ISSUE TITLE
including a play will cause load error for the role behind it

### ISSUE THREAD

**Issue by liuyong2000** [ansible/ansible#17662]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

ansible/lib/playbook/**init**.py  ???
##### ANSIBLE VERSION

```
ansible 2.1.0.0
```
##### CONFIGURATION

No change
##### OS / ENVIRONMENT

N/A
##### SUMMARY

including a play which has a relative path in a playbook, will cause load error for role in late play
##### STEPS TO REPRODUCE

pls follow the playbook below

```
- include: tasks/common/em2.yml
- name: play1
  hosts: SSHproxy
  remote_user: root
  gather_facts: false
  roles:
  - csfA

```

no specific requirement for the play included and role behind it.
main playbook, directory roles, and tasks directory are within same directory.

##### EXPECTED RESULTS

run this playbook correctly.
##### ACTUAL RESULTS

```
ERROR! the role 'csfA' was not found in /home/rover/ansible/tasks/common/roles:/home/rover/ansible/tasks/common:/etc/ansible/roles /home/rover/ansible/tasks/common

The error appears to have been in '/home/rover/ansible/inc.yml': line 8, column 5, but may
be elsewhere in the file depending on the exact syntax problem.
```

**Comment by liuyong2000** [comment]
I tried a fix and it passed my testing.
In finction Playbook:_load_playbook_data(self, file_name, variable_manager) which is in file ansible/playbook/__init__.py, when a playbook is included, this function creates a object of class PlaybookInclude, and pass it a parameter of loader, but after PlaybookInclude.load return, member _basedir in loader may be changed, this will impact the loading tasks/roles behind it.
So the fix is that a loader copy instead of reference is passed to PlaybookInclude.load.

```
     83         for entry in ds:
     84             if not isinstance(entry, dict):
     85                 raise AnsibleParserError("playbook entries must be either a valid play or an include statement", obj=entry)
     86 
     87             if 'include' in entry:
     88 +              import copy
     89 +              incl_load = copy.copy(self._loader)**
     90 -               pb = PlaybookInclude.load(entry, basedir=self._basedir, variable_manager=variable_manager, loader=self._loader)
     91 +               pb = PlaybookInclude.load(entry, basedir=self._basedir, variable_manager=variable_manager, loader=incl_load)
     92                 if pb is not None:
     93                     self._entries.extend(pb._entries)
     94                 else:
     95                     display.display("skipping playbook include '%s' due to conditional test failure" % entry.get('include', entry        ), color=C.COLOR_SKIP)
     96             else:
     97                 entry_obj = Play.load(entry, variable_manager=variable_manager, loader=self._loader)
     98                 self._entries.append(entry_obj)
```

**Comment by sivel** [comment]
This issue was resolved in 3fd94769dc0d74e06e26dcde049f47c26edb8286 and subsequently included in the v2.1.1 release.

If you have further questions please stop by IRC or the mailing list:

   * IRC: #ansible on irc.freenode.net
   * mailing list: https://groups.google.com/forum/#!forum/ansible-project
